nvm.sh

```diff
@@ -3688,11 +3872,29 @@ nvm() {
         nvm_sanitize_path nvm_has_colors nvm_process_parameters \
         nvm_node_version_has_solaris_binary nvm_iojs_version_has_solaris_binary \
         nvm_curl_libz_support nvm_command_info nvm_is_zsh nvm_stdout_is_terminal \
+        nvm_get_colors nvm_set_colors nvm_print_color_code \
+        nvm_echo_with_colors nvm_err_with_colors \
         >/dev/null 2>&1
       unset NVM_RC_VERSION NVM_NODEJS_ORG_MIRROR NVM_IOJS_ORG_MIRROR NVM_DIR \
         NVM_CD_FLAGS NVM_BIN NVM_INC NVM_MAKE_JOBS \
+        NVM_COLORS INSTALLED_COLOR SYSTEM_COLOR \
+        CURRENT_COLOR NOT_INSTALLED_COLOR DEFAULT_COLOR LTS_COLOR \
         >/dev/null 2>&1
     ;;
+    "set-colors")
+      local EXIT_CODE
+      nvm_set_colors "${1-}"
+      EXIT_CODE=$?
+      if [ "$EXIT_CODE" -eq 19 ]; then
+        >&2 nvm --help --no-colors
+        nvm_echo
+        nvm_err "Color is not supported on this system."
```

Comment:
it occurs to me that maybe it's ok to allow them to set the colors in this instance, as long as it warns them appropriately - eg, `TERM=dumb nvm set-colors` can fake it, and they might want to set the colors in a colorless subshell but still invoke the colors later.

I haven't convinced myself of this yet, but I wanted to write down the thought. What do yall think?

Reply:
We might have to research this more but it seems to be a more user-friendly option.

Reply:
Hi @ljharb, we've modified this in the latest commit [13a06bd](https://github.com/nvm-sh/nvm/pull/2291/commits/13a06bd3a8202a920785c9b9a9e3caeb2381996a). Please have a look and let us know if this could work.